src/sagemaker/local/image.py

```diff
@@ -147,22 +148,17 @@ def serve(self, primary_container):
         logger.info('creating hosting dir in {}'.format(self.container_root))
 
         model_dir = primary_container['ModelDataUrl']
-        if not model_dir.lower().startswith("s3://"):
-            for h in self.hosts:
-                host_dir = os.path.join(self.container_root, h)
-                os.makedirs(host_dir)
-                shutil.copytree(model_dir, os.path.join(self.container_root, h, 'model'))
-
+        volumes = self._prepare_serving_volumes(model_dir)
         env_vars = ['{}={}'.format(k, v) for k, v in primary_container['Environment'].items()]
 
-        _ecr_login_if_needed(self.sagemaker_session.boto_session, self.image)
-
         # If the user script was passed as a file:// mount it to the container.
-        script_dir = primary_container['Environment'][sagemaker.estimator.DIR_PARAM_NAME.upper()]
-        parsed_uri = urlparse(script_dir)
-        volumes = []
-        if parsed_uri.scheme == 'file':
-            volumes.append(_Volume(parsed_uri.path, '/opt/ml/code'))
+        if sagemaker.estimator.DIR_PARAM_NAME.upper() in primary_container['Environment']:
```

Comment:
What's the case where this is false?

Reply:
BYO :)  we only have sagemaker_submit_dir on our Frameworks.